orbitx/flight_gui.py

```diff
@@ -125,6 +90,260 @@ def callback(*_):
         self.recentre_camera(DEFAULT_CENTRE)
     # end of __self__
 
+    ####################### pritvate methods to initial settings ############################
+
+    # def _vpython_version_check(self) -> None :
+    #     ctrl_c_handler: function = signal.getsignal(signal.SIGINT)
+    #     if self._vpython.__version__ == '7.4.7':
+    #         # vpython installs "os._exit(0)" signal handlers in 7.4.7.
+    #         # Very not good for us.
+    #         # Reset the signal handler to default behaviour.
+    #         signal.signal(signal.SIGINT, ctrl_c_handler)
+
+    #         # os._exit(0) is in one other place, let's take that out too.
+    #         # Unfortunately, now that vpython isn't calling os._exit, we have
+    #         # to notify someone when the GUI closed.
+    #         # Note to access vpython double underscore variables, we have to
+    #         # bypass normal name mangling by using getattr.
+    #         def callback(*_):
+    #             getattr(self._vpython.no_notebook, '__interact_loop').stop()
+    #             self.closed = True
+    #         # end of callback
+    #         self.closed = False
+    #         getattr(self._vpython.no_notebook,'__factory').protocol.onClose = callback
+    # # end of _vpython_version_check
+
+    def _init_canvas(self) -> vpython.vpython.canvas :
```

Comment:
Why is `_init_canvas` in its own function, if it only ever gets called once in \_\_init__? It seems to me the code to separate this just separates code from where it's used, but you may know something more than me

Reply:
actually, it makes more sense if _init_canvas is it's own function now that I think about it. Thank you for doing that (y)